src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -0,0 +1,458 @@
+    return ret;
+}
+
+int32_t AndroidCryptoNative_X509GetThumbprint(jobject /*X509Certificate*/ cert, uint8_t *buf, int32_t len)
+{
+    assert(cert != NULL);
+    JNIEnv *env = GetJNIEnv();
+
+    int32_t ret = BUFFER_FAIL;
+    INIT_LOCALS(loc, algorithm, md, encoded, thumbprint)
+
+    // MessageDigest md = MessageDigest.getInstance("SHA-1");
+    loc[algorithm] = JSTRING("SHA-1");
+    loc[md] = (*env)->CallStaticObjectMethod(env, g_mdClass, g_mdGetInstanceMethod, loc[algorithm]);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+
+    // byte[] encoded = cert.getEncoded();
+    loc[encoded] = (*env)->CallObjectMethod(env, cert, g_X509CertGetEncoded);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+
+    // byte[] thumbprint = md.digest(encoded);
+    // return thumbprint.length;
+    loc[thumbprint] = (*env)->CallObjectMethod(env, loc[md], g_mdDigestMethod, loc[encoded]);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+    ret = PopulateByteArray(env, loc[thumbprint], buf, len);
+
+cleanup:
+    RELEASE_LOCALS(loc, env)
+    return ret;
+}
+
+int32_t AndroidCryptoNative_X509GetVersion(jobject /*X509Certificate*/ cert)
+{
+    assert(cert != NULL);
+    JNIEnv *env = GetJNIEnv();
+
+    // return cert.getVersion();
+    jint ver = (*env)->CallIntMethod(env, cert, g_X509CertGetVersion);
+    return CheckJNIExceptions(env) ? VALUE_FAIL : (int32_t)ver;
+}
+
+static int32_t GetNameBytes(JNIEnv *env, jobject /*X500Principal*/ name, uint8_t *buf, int32_t len)
+{
+    assert(name != NULL);
+    int32_t ret = BUFFER_FAIL;
+
+    // byte[] raw = name.getEncoded();
+    // return raw.length
+    jbyteArray encoded = (*env)->CallObjectMethod(env, name, g_X500PrincipalGetEncoded);
+    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+    ret = PopulateByteArray(env, encoded, buf, len);
+
+cleanup:
+    (*env)->DeleteLocalRef(env, encoded);
+    return ret;
+}
+
+int32_t AndroidCryptoNative_X509GetIssuerNameBytes(jobject /*X509Certificate*/ cert, uint8_t *buf, int32_t len)
+{
+    assert(cert != NULL);
+    JNIEnv *env = GetJNIEnv();
+    int32_t ret = BUFFER_FAIL;
+
+    // X500Principal name = cert.getIssuerX500Principal()
+    jobject name = (*env)->CallObjectMethod(env, cert, g_X509CertGetIssuerX500Principal);
+    if (!CheckJNIExceptions(env) && name != NULL)
+        ret = GetNameBytes(env, name, buf, len);
+
+    (*env)->DeleteLocalRef(env, name);
+    return ret;
+}
+
+int32_t AndroidCryptoNative_X509GetSubjectNameBytes(jobject /*X509Certificate*/ cert, uint8_t *buf, int32_t len)
+{
+    assert(cert != NULL);
+    JNIEnv *env = GetJNIEnv();
+    int32_t ret = BUFFER_FAIL;
+
+    // X500Principal name = cert.getSubjectX500Principal()
+    jobject name = (*env)->CallObjectMethod(env, cert, g_X509CertGetSubjectX500Principal);
+    if (!CheckJNIExceptions(env) && name != NULL)
+        ret = GetNameBytes(env, name, buf, len);
+
+    (*env)->DeleteLocalRef(env, name);
+    return ret;
+}
+
+uint64_t AndroidCryptoNative_X509IssuerNameHash(jobject /*X509Certificate*/ cert)
```

Comment:
I think we only use this for our CRL cache.  Assuming that you're just calling Android API for the cert chain stuff you probably don't need this at all.